### PR TITLE
Add aten::_ctc_loss and its variants

### PR DESCRIPTION
**Pull request by xytintel**
- [x] _ctc_loss
- [x] _ctc_loss.Tensor
- [x]  _ctc_loss_backward
- [x] _ctc_loss_backward.Tensor

### CODE REVIEW

**Review by dvrogozh — CHANGES_REQUESTED** [review]
ops should be excluded from here:
https://github.com/intel/torch-xpu-ops/blob/c13499159c2a8d7b8f67342674422b165b24684f/src/ATen/native/xpu/XPUFallback.template#L169-L170
and added to
https://github.com/intel/torch-xpu-ops/blob/main/yaml/xpu_functions.yaml

File: src/ATen/native/xpu/sycl/LossCTCKernels.cpp [hunk 1]
```diff
@@ -0,0 +1,1266 @@
...
+      at::tensor(input_lengths, targets.options().dtype(kLong));
+  tg_batch_offsets = tg_batch_offsets.to(targets.device());
+
+  Tensor log_alpha = at::empty(
+      {batch_size, log_probs.size(0), 2 * max_target_length + 1},
+      log_probs.options());
+  Tensor neg_log_likelihood = at::empty({batch_size}, log_probs.options());
+
+  // Very likely, we could be more clever here, e.g. learning (or generalizing
+  // and reusing) from SoftMax.cu...
```
Comment (fengyuan14) [inline src/ATen/native/xpu/sycl/LossCTCKernels.cpp:373]: The comments here do seem CUDA related. Please correct and elaborate the comments from Intel GPU perspective.

File: src/ATen/native/xpu/sycl/LossCTCKernels.cpp [hunk 2]
```diff
@@ -0,0 +1,1266 @@
...
+  constexpr int max_threads = std::is_same<scalar_t, float>::value
+      ? 1024
```
Comment (fengyuan14) [inline src/ATen/native/xpu/sycl/LossCTCKernels.cpp:375]: Which is not device compatible, if simd16 and LGRF are enabled on PVC or SIMD8 is enabled on MTL, please use runtime query.

File: src/ATen/native/xpu/sycl/LossCTCKernels.cpp [hunk 3]
```diff
@@ -0,0 +1,1266 @@
...
+
+  Tensor grad = at::full_like(
+      log_probs,
+      neginf,
+      LEGACY_CONTIGUOUS_MEMORY_FORMAT); // initialization for log(sum (alpha
+                                        // beta))
+
+  // As above, there may be better configurations to use.
+  constexpr int max_threads = std::is_same<scalar_t, float>::value
+      ? 1024
```
Comment (fengyuan14) [inline src/ATen/native/xpu/sycl/LossCTCKernels.cpp:1010]: Ditto.

**Comment by xytintel** [comment]
image
image

**Review by ZhiweiYan-96 — APPROVED** [review]

**Pull request merged**